middlewares/ip/ip_whitelist.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package ip
+
+import (
+	"net"
+
+	"github.com/chihaya/chihaya"
+	"github.com/chihaya/chihaya/config"
+	"github.com/chihaya/chihaya/server/store"
+	"github.com/chihaya/chihaya/tracker"
+)
+
+func init() {
+	tracker.RegisterAnnounceMiddleware("ip_whitelist", whitelistAnnounceIP)
+}
+
+// whitelistAnnounceIP provides a middleware that only allows IPs to announce that are stored in an IPStore
+func whitelistAnnounceIP(next tracker.AnnounceHandler) tracker.AnnounceHandler {
+	return func(cfg *config.TrackerConfig, req *chihaya.AnnounceRequest, resp *chihaya.AnnounceResponse) error {
+		var (
+			whitelisted bool
+			err         error
```

Comment:
you can save some lines if you change the signature of the function to return `(err error)` and then this block turns into just a `var whitelisted bool` or a `whitelisted := false`